### DISCUSSION TITLE
gomplate replace multiple strings in yaml

### DISCUSSION THREAD

**Discussion by HenryXie1** [hairyhenderson/gomplate#1241] [Q&A] [answered]
@hairyhenderson  Thanks for answering https://github.com/hairyhenderson/gomplate/discussions/1239
I met another questions when I use gomplate to replace CA pem data in a yaml
```
mytest:
    certficate-auth:
         MIIsdfefefwfw...
         DIEKFOEKFIE....
         FEWFEWFEF.....
```
When I use  gomplate to replace data (multiple lines) with new data, it does not know about space indent of yaml, the result is like
```
mytest:
    certficate-auth:
         NEWDATADKDKKD...
FEFWEFE....
FEWEFEF....
```

I try to use `regexp.Replace "^" "          "` , but it only replace the first line
Can you give some clues how to handle that? tks

**Comment by hairyhenderson** [comment, marked as answer]
You want the `strings.Indent`/`indent` function: https://docs.gomplate.ca/functions/strings/#strings-indent
